include/odp/api/spec/system_info.h

```diff
@@ -191,6 +194,70 @@ typedef struct odp_system_info_t {
 
 } odp_system_info_t;
 
+/**
+ * Memory information
+ */
+typedef struct odp_system_meminfo_t {
+	/**
+	 * Total mapped memory
+	 *
+	 * Total amount of memory (in bytes) in all memory pages that are reserved by
+	 * this ODP instance from the system.
+	 */
+	uint64_t total_mapped;
+
+	/**
+	 * Total memory usage
+	 *
+	 * Total amount of memory (in bytes) that is currently in use by this ODP instance.
+	 * This is a subset of 'total_mapped' bytes.
+	 */
+	uint64_t total_used;
+
+	/**
+	 * Total memory usage overheads
+	 *
+	 * Total amount of memory (in bytes) that is currently consumed by roundings to
+	 * alignment/block/page size limits, etc. overheads. This is a subset of 'total_used'
+	 * bytes.
+	 */
+	uint64_t total_overhead;
+
+} odp_system_meminfo_t;
+
+/**
+ * Memory block information
+ */
+typedef struct odp_system_memblock_t {
+	/** Memory block name */
+	char name[ODP_SYSTEM_MEMBLOCK_NAME_LEN];
+
+	/** Start address of the block */
+	uintptr_t addr;
+
+	/**
+	 * Memory usage
+	 *
+	 * Total amount of memory (in bytes) that is used by this block.
+	 */
+	uint64_t used;
+
+	/**
+	 * Memory usage overheads
+	 *
+	 * Total amount of memory (in bytes) that is currently consumed by roundings to
+	 * alignment/block/page size limits, etc. overheads. This is a subset of 'used' bytes.
+	 */
+	uint64_t overhead;
```

Comment:
What will be the use case of identifying the overheads? 
(used - overheads) will give actually memory demanded by user, but (used) was allocated.

Reply:
Yes, used - overhead is the memory actually requested by user or needed by an implementation internal memory struct. A simple implementation (like linux-gen SHM) may e.g. always allocate a new page(s) for every memory block reservation request, which waste a lot of memory if there are many small allocations (e.g. if a 12B request reserves 4kB -> 4084B wasted as overhead). From these statistics a user may see where the largest wastes are, and may tune/combine allocations, etc.

Reply:
sure. Thanks for explanation.